### PR TITLE
devops templates

### PR DESCRIPTION
**Pull request by babicahmed**
- Refactoring of existing setup
- New template for tfsec
- Infrascan budget check

### CODE REVIEW

File: templates/new_structure/BuildStage.yml [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+  default: 'none'
+- name: workingDirectory
+  type: string
+- name: checkovTestResultsFileName
+  type: string
+- name: dependsOn
+  type: string
+- name: environment
+  type: string
+#- name: privatePoolName
```
Comment (tpeterson66) [inline templates/new_structure/BuildStage.yml]: Why did this get removed? I think this is something we're going to need.

File: templates/new_structure/DeployStage.yml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+stages:
+- stage: apply
+  condition: |
+    and (
+      succeeded(),
+      in(dependencies.build.result, 'Succeeded')
+    )
+  dependsOn:
+    - "build"
+  displayName: "Apply"
```
Comment (tpeterson66) [inline templates/new_structure/DeployStage.yml:31]: We might need support for private agents - lets default to using the azure provided or default, but the option to use private agents.

File: templates/new_structure/Pipeline.yml [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+      workingDirectory: '$(System.DefaultWorkingDirectory)/terraform'
+      checkovImageName: 'bridgecrew/checkov'
+      checkovTestResultsFileName: 'CheckovReport.xml' 
+      scanDirectory: 'terraform'
+      terrascanImageName: 'accurics/terrascan'
+      tfsecImageName: 'aquasec/tfsec'
+      skipCheck: none
+      dependsOn: 'runCheckov'
+      environment: 'dev'
+#        privatePoolName: 'cspmigtest2-build-agents'
```
Comment (tpeterson66) [inline templates/new_structure/Pipeline.yml]: Same comment as before, but we're going to need to support this.

File: templates/new_structure/autoApply.yml [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+                backendAzureRmContainerName: ${{ parameters.backendContainerName }}
+                backendAzureRmKey: ${{ parameters.environment }}.tfstate
+              displayName: "Run > terraform init"
+            - task: TerraformTaskV3@3
+              inputs:
+                command: "apply"
+                workingDirectory: ${{ parameters.workingDirectory }}
+                environmentServiceNameAzureRM: ${{ parameters.serviceConnection }}
+                publishPlanResults: "PlanResults"
+      #          commandOptions: "-var backendAzureRmResourceGroupName=${{ parameters.backendResourceGroupName }} -var backendAzureRmStorageAccountName=${{ parameters.backendStorageAccountName }} -var backendAzureRmContainerName=${{ parameters.backendContainerName }} -var backendAzureRmKey=${{ parameters.environment }}.tfstate -var-file=${{ parameters.workingDirectory }}/variables/${{ parameters.environment }}.auto.tfvars"
```
Comment (tpeterson66) [inline templates/new_structure/autoApply.yml]: Why did this get commented out?

**Review by tpeterson66 — CHANGES_REQUESTED** [review]
Lets setup a call to chat about the templates and how we can test a few different use-cases.

File: templates/new_structure/BuildStage.yml (hunk 1 above)
Comment (babicahmed) [reply templates/new_structure/BuildStage.yml]: Fixed in new commits.

File: templates/new_structure/DeployStage.yml (hunk 2 above)
Comment (babicahmed) [reply templates/new_structure/DeployStage.yml:31]: Fixed in new commits.

File: templates/new_structure/Pipeline.yml (hunk 3 above)
Comment (babicahmed) [reply templates/new_structure/Pipeline.yml]: Fixed in new commits.

File: templates/new_structure/autoApply.yml (hunk 4 above)
Comment (babicahmed) [reply templates/new_structure/autoApply.yml]: Updated and tested. Fixed in new commits.

**Review by tpeterson66 — APPROVED** [review]

**Pull request merged**